metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/dao/DatasetDao.java

```diff
@@ -410,11 +421,18 @@ public String checkAndCreateDatasetInEcloud(Dataset dataset) throws ExternalTask
     if (!criteriaContainerGroups.isEmpty()) {
       query.or(criteriaContainerGroups.toArray(new CriteriaContainer[0]));
     }
-    query.order(DATASET_ID.getFieldName());
+    query.order(Sort.ascending(DATASET_ID.getFieldName()));
 
-    return ExternalRequestUtil.retryableExternalRequestConnectionReset(() -> query.asList(
-        new FindOptions().skip(nextPage * getDatasetsPerRequest())
-            .limit(getDatasetsPerRequest())));
+    final FindOptions findOptions = new FindOptions().skip(nextPage * getDatasetsPerRequest())
+        .limit(getDatasetsPerRequest());
+    return ExternalRequestUtil
+        .retryableExternalRequestConnectionReset(() -> getListOfQuery(query, findOptions));
+  }
+
+  private <T> List<T> getListOfQuery(Query<T> query, FindOptions findOptions) {
+    try (MorphiaCursor<T> cursor = query.find(findOptions)) {
+      return cursor.toList();
+    }
   }
```

Comment:
Sorry if my comment was not clear. I think you should do the same for this one as well: move the 'ExternalRequestUtil.retryableExternalRequestConnectionReset' to this private method.